plugins/resource_monitor_plugin/include/eosio/resource_monitor_plugin/file_space_handler.hpp

```diff
@@ -53,16 +57,15 @@ namespace eosio::resource_monitor {
                continue;
             }
 
-            if ( info.available < fs.warning_available ) {
-               wlog("Space usage on ${path}'s file system approaching threshold. available: ${available}, warning_available: ${warning_available}", ("path", fs.path_name.string()) ("available", info.available) ("warning_available", fs.warning_available));
-               if ( shutdown_on_exceeded ) {
-                  wlog("nodeos will shutdown when space usage exceeds threshold ${threshold}%", ("threshold", shutdown_threshold));
+            if ( info.available < fs.shutdown_available ) {
+               if (output_threshold_warning) {
+                  wlog("Space usage warning: ${path}'s file system exceeded threshold ${threshold}%, available: ${available}, Capacity: ${capacity}, shutdown_available: ${shutdown_available}", ("path", fs.path_name.string()) ("threshold", shutdown_threshold) ("available", info.available) ("capacity", info.capacity) ("shutdown_available", fs.shutdown_available));
                }
-
-               if ( info.available < fs.shutdown_available ) {
-                  wlog("Space usage on ${path}'s file system exceeded threshold ${threshold}%, available: ${available}, Capacity: ${capacity}, shutdown_available: ${shutdown_available}", ("path", fs.path_name.string()) ("threshold", shutdown_threshold) ("available", info.available) ("capacity", info.capacity) ("shutdown_available", fs.shutdown_available));
-
-                  return true;
+               return true;
+            } else if ( info.available < fs.warning_available && output_threshold_warning ) {
```

Comment:
why you are not returning ```true``` if ```available``` is less than ```warning_available``` ?

Reply:
The current process is: only when `available` is less than `shutdown_available`, this method will return `true`; otherwise, if `available` is less than `warning_available`, it will not return but only output warnings. Name like `is_shutdown_threshold_exceeded` may be more accurate, do you think it's necessary to modify the name?

Reply:
I missed that. Apologize. I think you can leave old code name.
Also in original code it assumed that shutdown limit is always lower than warning so it is checked only if warning limit reached. Are you sure this logic is not broken? if statements reorganization looks a bit confusing, can you preserve original structure and just add another if statement(s) inside or add more to existing if statement(s)? Wouldn't it be easier?

Reply:
Yes, shutdown limit is always lower than warning. My consideration is: in the original code, when `if (info.available < fs.shutdown_available)` is within `if (info.available < fs.warning_available)`, and if the shutdown threshold is hit, warning message (`...file system approaching threshold...`) will be (unnecessarily) output and immediately followed by the shutdown warning (`...file system exceeded threshold`). I think here the first warning message is unnecessary. But maybe I missed something? Thanks.

Reply:
if there an assert to check that ```shutdown_available``` is always less than ```warning_available``` that makes sense. this part looks good than.

Reply:
Yes, it's hard-coded that `shutdown_available` is less than `warning_available` (which are calculated from `shutdown_threshold` and `warning_threshold`, and `shutdown_threshold - warning_threshold = 5 (space_threshold_warning_diff)`).